contracts/strategy/rysk/RyskStrategy.sol

```diff
@@ -0,0 +1,234 @@
+    }
+
+    //
+    // IStrategy
+    //
+
+    /**
+     * Rysk strategy is asynchronous meaning it doesn't support immediate withdrawals.
+     *
+     * @return false always
+     */
+    function isSync() external pure override(IStrategy) returns (bool) {
+        return false;
+    }
+
+    /// @inheritdoc IStrategy
+    function hasAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (bool)
+    {
+        return _getTotalShares() != 0;
+    }
+
+    /// @inheritdoc IStrategy
+    function investedAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        return _sharesToUnderlying(_getTotalShares());
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        emit StrategyInvested(balance);
+
+        ryskLqPool.deposit(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    function withdrawToVault(uint256 _amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (_amount == 0) revert StrategyAmountZero();
+
+        uint256 shares = _underlyingToShares(_amount);
+
+        _resetPendingWithdrawalIfEpochAdvanced();
+
+        // because the vault can call updateInvested => withdrawToVault multiple times during the same epoch,
+        // and rysk liquidity pool aggregates withdrawal amounts inititated during the same epoch,
+        // we cannot withdraw less than already pending amount
+        if (pendingWithdrawal.shares >= shares) return;
+
+        uint256 sharesToWithdraw = shares - pendingWithdrawal.shares;
+        if (!_hasEnoughShares(sharesToWithdraw))
+            revert StrategyNotEnoughShares();
+
+        pendingWithdrawal.shares += uint128(sharesToWithdraw);
+
+        emit RyskWithdrawalInitiated(sharesToWithdraw);
+        ryskLqPool.initiateWithdraw(sharesToWithdraw);
+    }
+
+    /**
+     * Completes the withdrawal initiated by withdrawToVault.
+     *
+     * @notice Expected to be called by the backend once the Rysk liquidity pool enters a new epoch.
+     * Backend should subscribe to 'EpochExecuted' event on Rysk LiquidityPool contract,
+     * and act by calling this function when the event is emitted to complete ongoing withdrawal.
+     */
+    function completeWithdrawal() external {
+        if (pendingWithdrawal.epoch == 0) revert RyskNoWithdrawalInitiated();
+        if (pendingWithdrawal.epoch == ryskLqPool.epoch())
+            revert RyskCannotComipleteWithdrawalInSameEpoch();
+
+        uint256 sharesToWithdraw = ryskLqPool
+            .withdrawalReceipts(address(this))
+            .shares;
+
+        uint256 amountWithdrawn = ryskLqPool.completeWithdraw(sharesToWithdraw);
```

Comment:
this is fine now, but in the older version u would have needed to check the withdrawal amount to make sure the pool can honour it. Now the lpool always honours it.